phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -342,8 +343,10 @@ public Scan intersectScan(Scan scan, final byte[] originalStartKey, final byte[]
             // TODO: it seems that our SkipScanFilter or HBase runs into problems if we don't
             // have an enclosing range when we do a point lookup.
             if (isPointLookup) {
-                scanStartKey = ScanUtil.getMinKey(schema, newSkipScanFilter.getSlots(), slotSpan);
-                scanStopKey = ScanUtil.getMaxKey(schema, newSkipScanFilter.getSlots(), slotSpan);
+                scanStartKey = ScanUtil.getMinKey(
+                        newSkipScanFilter.getSchema(), newSkipScanFilter.getSlots(), newSkipScanFilter.getSlotSpan());
```

Comment:
Why we made this change? Whats' the diff btw 'RowKeySchema' from the `ScanRangers` and `SkipScanFilter.getSchema()`?

Reply:
When the PK column type is varchar, the column is translated to multiple char[1] columns in SkipScanFilter's schema, and the SkipScanFilter's schema is supposed to use here, otherwise it will cause the test failures in SkipScanFilterIntersectTest.java.